go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -41,11 +41,7 @@ func (npw *nodePortWatcher) AddService(service *kapi.Service) {
 	for _, svcPort := range service.Spec.Ports {
 		protocol := strings.ToLower(string(svcPort.Protocol))
 		protocol6 := protocol + "6"
-		if util.ServiceTypeHasNodePort(service) {
-			if err := util.ValidatePort(svcPort.Protocol, svcPort.NodePort); err != nil {
```

Comment:
If you're removing all `ValidatePort` in `pkg/node/*` why are you not changing all files (`gateway_localnet.go`/`gateway_iptables.go`, etc) too?

Reply:
because I cherry picked this changes from the dual stack PR, to try to break it down ... and those files need more changes for dual stack 
https://github.com/ovn-org/ovn-kubernetes/pull/1857